src/modules/sync/lib/tree-utils.ts

```diff
@@ -0,0 +1,79 @@
+import { TreeNode } from 'src/components/tree-view';
+import { SYNC_EXCLUSIONS } from '../constants';
+import type { RawFileNode } from '../types';
+
+export const shouldExcludeFromSync = ( itemName: string ): boolean => {
+	if ( itemName.startsWith( '.' ) ) {
+		return true;
+	}
+
+	if ( SYNC_EXCLUSIONS.includes( itemName ) ) {
+		return true;
+	}
+
+	return false;
+};
+
+export const shouldLimitDepth = ( relativePath: string ): boolean => {
+	const normalizedPath = relativePath.replace( /^wp-content\//, '' );
+
+	// Match plugins/plugin-name or plugins/plugin-name/
+	if ( normalizedPath.match( /^plugins\/[^/]+\/?$/ ) ) {
+		return true;
+	}
+
+	// Match themes/theme-name or themes/theme-name/
+	if ( normalizedPath.match( /^themes\/[^/]+\/?$/ ) ) {
+		return true;
+	}
+
+	return false;
+};
+
+export const convertRawToTreeNodes = (
+	rawNodes: RawFileNode[],
+	parentChecked: boolean = false
+): TreeNode[] => {
+	const pluginRegex = /^plugins\/[^/]+$/;
+	const themeRegex = /^themes\/[^/]+$/;
+	const pathCleanRegex = /[^a-zA-Z0-9]/g;
+	return rawNodes
+		.map( ( rawNode ) => {
+			let nodeType: 'file' | 'folder' | 'plugin' | 'theme' = rawNode.isDirectory
+				? 'folder'
+				: 'file';
+			if ( rawNode.isDirectory ) {
+				const normalizedPath = rawNode.path.replace( /^wp-content\//, '' );
+				if ( pluginRegex.test( normalizedPath ) ) {
+					nodeType = 'plugin';
+				} else if ( themeRegex.test( normalizedPath ) ) {
+					nodeType = 'theme';
+				}
+			}
+
+			const children = rawNode.children
+				? convertRawToTreeNodes( rawNode.children, parentChecked )
+				: rawNode.isDirectory
+				? []
+				: undefined;
```

Comment:
```suggestion
			let children: TreeNode[] | undefined;
			if ( rawNode.children ) {
				children = convertRawToTreeNodes( rawNode.children, parentChecked );
			} else if ( rawNode.isDirectory ) {
				children = [];
			} else {
				children = undefined;
			}
```
Let's unwrap the nested ternary here.

Reply:
https://github.com/Automattic/studio/pull/1940/commits/2fa32c64770cf4a1e9d334d158baf2dbe4446c9d